src/acl.c

```diff
@@ -1937,6 +1966,68 @@ sds *ACLMergeSelectorArguments(sds *argv, int argc, int *merged_argc, int *inval
     return acl_args;
 }
 
+/* takes an acl string already split on spaces and adds it to the given user
+ * if the user object is NULL, will create a user with the given username
+ *
+ * Returns an error as an sds string if the ACL string is not parsable
+ */
+sds ACLStringSetUser(user *u, sds username, sds *argv, int argc) {
+    serverAssert(u != NULL || username != NULL);
+
+    sds error = NULL;
+
+    int merged_argc = 0, invalid_idx = 0;
+    sds *acl_args = ACLMergeSelectorArguments(argv, argc, &merged_argc, &invalid_idx);
+
+    if (!acl_args) {
+        return sdscatfmt(sdsempty(),
+                         "Unmatched parenthesis in acl selector starting "
+                         "at '%.*s'.", sdslen(argv[invalid_idx]), (char *) argv[invalid_idx]);
```

Comment:
don't know, think I was conflating multiple different things (thinking its an sds so have to use len, but then I should have just used S).  changed it to how it was.

Reply:
i'm still afraid of protocol violation here.
in case someone executes an ACL command with newline as part of one of the args, it'll respond with a text that includes that arg, and since it doesn't go now though the newline trimming of `addReplyErrorFormatInternal`, it can mess up the protocol.
maybe we need to create a new addReplyErrorSdsSafe, and use it instead of addReplyErrorSds

Reply:
@sjpotter ^^^

Reply:
ok, would returning the errors like this be better

```
        sds err = sdsnew("Unmatched parenthesis in acl selector starting at ");
        return sdscatrepr(err, argv[invalid_idx], sdslen(argv[invalid_idx]));
```

Reply:
I believe I now handle it better - addReplyErrorSdsSafe() that does the mapping on reply.